addons/mass_mailing/static/src/js/mass_mailing_form.js

```diff
@@ -1,10 +1,107 @@
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        bus.off('DOM_updated', this, this._boundTriggerIframeUpdated);
+        bus.off('iframe_updated', this, this._boundInitIframeResizer);
+        this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Resize the given iframe so its height fits its contents and initialize a
+     * resize observer to resize on each size change in its contents.
+     *
+     * @param {JQuery} $iframe
+     */
+    _initIframeResizer($iframe) {
+        if (!$iframe.length || !$iframe.contents().length) {
+            return;
+        }
+        this._autoresizeIframe($iframe);
+        const $iframeDoc = $iframe.contents();
+        const $ref = $iframeDoc.find('#iframe_target');
+        if ($ref[0]) {
+            const isFull = window.top.document.body.classList.contains("o_field_widgetTextHtml_fullscreen");
+            $ref.css({
+                display: isFull ? '' : 'flex',
+                'flex-direction': isFull ? '' : 'column',
+            });
+            if (this._resizeObserver) {
+                this._resizeObserver.disconnect();
+            }
+            this._resizeObserver = new ResizeObserver(() => this._autoresizeIframe($iframe));
+            this._resizeObserver.observe($ref[0]);
+        }
+        const $content = this.$('.o_content');
+        $content.on('scroll', () => this._repositionSidebar($iframe));
+        const $editable = $iframeDoc.find('.note-editable');
+        const wysiwyg = $editable.data('wysiwyg');
+        $iframeDoc.find('body').on('click', '.o_fullscreen_btn', () => {
+            const isFull = window.top.document.body.classList.contains("o_field_widgetTextHtml_fullscreen");
+            $ref.css({
+                display: isFull ? '' : 'flex',
+                'flex-direction': isFull ? '' : 'column',
+            });
+            if (wysiwyg && wysiwyg.snippetsMenu) {
+                // Restore the appropriate scrollable depending on the mode.
+                this._$scrollable = this._$scrollable || wysiwyg.snippetsMenu.$scrollable;
+                wysiwyg.snippetsMenu.$scrollable = isFull ? $iframeDoc.find('.note-editable') : this._$scrollable;
+            }
+            $content.trigger('scroll');
+        });
+    },
+    /**
+     * Trigger an 'iframe_updated' event on a bus, passing it the mass mailing
+     * iframe jquery element.
+     */
+    _triggerIframeUpdated() {
+        bus.trigger('iframe_updated', this.$('iframe.wysiwyg_iframe, iframe.o_readonly'));
+    },
+    /**
+     * Resize the given iframe so its height fits its contents.
+     *
+     * @param {JQuery} $iframe
+     */
+    _autoresizeIframe($iframe) {
+        const minHeight = $(window).height() - Math.abs($iframe.offset().top) - 6;
```

Comment:
Lots of magic numbers (6, 12 here below), makes the code a bit difficult to understand, would be nice to add small comments to what they are referring to.